integrations/async-gitlab/.port/resources/blueprints.json

```diff
@@ -0,0 +1,228 @@
+          "url": {
+            "title": "URL",
+            "type": "string",
+            "format": "url"
+          },
+          "readme": {
+            "title": "README",
+            "type": "string",
+            "format": "markdown"
+          },
+          "description": {
+            "title": "Description",
+            "type": "string"
+          },
+          "language": {
+            "title": "Language",
+            "type": "string"
+          },
+          "namespace": {
+            "title": "Namespace",
+            "type": "string"
+          },
+          "fullPath": {
+            "title": "Full Path",
+            "type": "string"
+          },
+          "defaultBranch": {
+            "title": "Default Branch",
+            "type": "string"
+          }
+        },
+        "required": []
+      },
+      "mirrorProperties": {},
+      "calculationProperties": {},
+      "relations": {
+        "group": {
+          "title": "Group",
+          "target": "gitlabGroup",
+          "required": true,
+          "many": false
+         }
+       }
+    },
+    {
+      "identifier": "gitlabMergeRequest",
+      "title": "Merge Request",
+      "icon": "GitVersion",
+      "schema": {
+        "properties": {
+          "creator": {
+            "title": "Creator",
+            "type": "string"
+          },
+          "status": {
+            "title": "Status",
+            "type": "string",
+            "enum": [
+              "opened",
+              "closed",
+              "merged",
+              "locked"
+            ],
+            "enumColors": {
+              "opened": "yellow",
+              "closed": "red",
+              "merged": "green",
+              "locked": "blue"
+            }
+          },
+          "createdAt": {
+            "title": "Created At",
+            "type": "string",
+            "format": "date-time"
+          },
+          "updatedAt": {
+            "title": "Updated At",
+            "type": "string",
+            "format": "date-time"
+          },
+          "mergedAt": {
+            "title": "Merged At",
+            "type": "string",
+            "format": "date-time"
+          },
+          "link": {
+            "title": "Link",
+            "format": "url",
+            "type": "string"
+          },
+          "reviewers": {
+            "title": "Reviewers",
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          }
+        },
+        "required": []
+      },
+      "mirrorProperties": {},
+      "calculationProperties": {
+        "lead_time_days": {
+          "title": "Lead time (Days)",
+          "calculation": "if .properties.status == \"merged\" then ((.properties.mergedAt | sub(\"\\\\.[0-9]+\\\\+00:00$\"; \"Z\") | strptime(\"%Y-%m-%dT%H:%M:%SZ\") | mktime) - (.properties.createdAt | sub(\"\\\\.[0-9]+\\\\+00:00$\"; \"Z\") | strptime(\"%Y-%m-%dT%H:%M:%SZ\") | mktime)) / 86400 | tonumber else null end",
```

Comment:
why is lead time a calculation property here, The calculation should be done in the mapping configuration and default to null if the pr hasn't been merged. we have access to both mergedAt and createdAt at the time of ingesting. this would help the reuse of leadTime. calculation properties are done on the fly. there are instances that we need to use leadTime.